numeric.c

```diff
@@ -1675,14 +1675,18 @@ Perl_my_atof3(pTHX_ const char* orig, NV* value, const STRLEN len)
             return (char *)s+1;
         }
 
+        /* strtod will parse a sign (and skip leading whitespaces) by itself,
+         * so rewind s to the beginning of the string. */
```

Comment:
I think it is important to point out is the reason we _should_ let it parse them itself, ie that we'd get an undesirable result if we pointed it at some apparently-leading whitespace _after_ the sign; maybe something like this:
```
        /* We do not want strtod to parse whitespace after the sign, since
         * that would give backward-incompatible results. So we rewind and 
         * let strtod handle the whitespace and sign character itself. */
```

Reply:
Thank you, but I've forgotten that `"+-inf"` is still wrongly numified, so I will fix it too.